inc/bdx86_core.h

```diff
@@ -347,15 +347,13 @@ typedef ND_UINT32 ND_REG_SIZE;
 // Misc macros.
 //
 
-// Sign extend 8 bit to 64 bit.
-#define ND_SIGN_EX_8(x)             (((x) & 0x00000080) ? (0xFFFFFFFFFFFFFF00 | (x)) : ((x) & 0xFF))
-// Sign extend 16 bit to 64 bit.
-#define ND_SIGN_EX_16(x)            (((x) & 0x00008000) ? (0xFFFFFFFFFFFF0000 | (x)) : ((x) & 0xFFFF))
-// Sign extend 32 bit to 64 bit.
-#define ND_SIGN_EX_32(x)            (((x) & 0x80000000) ? (0xFFFFFFFF00000000 | (x)) : ((x) & 0xFFFFFFFF))
-// Wrapper for for ND_SIGN_EX_8/ND_SIGN_EX_16/ND_SIGN_EX_32. Sign extend sz bytes to 64 bits.
-#define ND_SIGN_EX(sz, x)           ((sz) == 1 ? ND_SIGN_EX_8(x) : (sz) == 2 ? ND_SIGN_EX_16(x) :                   \
-                                     (sz) == 4 ? ND_SIGN_EX_32(x) : (x))
+#define ND_UINT_SZ_MASK(sz)         ( ((ND_UINT64) -1) >> ( ( (8 - (sz) ) & 0x7 ) * 8) )
+
+// Sign extend to 64 bit, with minimal branches
+#define ND_SIGN_EX(sz, x)           ( ( (x) & (0x8000000000000000ULL >> ( ((8 - (sz) ) & 0x7 ) * 8) ) )             \
```

Comment:
There is no need for conditions - this could be simplified by writing it as follows:
```
#define ND_SIGN_EX(sz, x)          (((x) & ND_SIZE_TO_MASK(sz)) | (~ND_SIZE_TO_MASK(sz) * ND_GET_SIGN(sz, x)))
```

Reply:
You're right, this is better.